src/Dialogs/MainWindow.vala

```diff
@@ -39,27 +39,26 @@ public class Tootle.Dialogs.MainWindow: Hdy.Window, ISavedWindow {
 		return view;
 	}
 
-	public bool back () {
-		var children = deck.get_children ();
-		unowned var current = children.find (deck.visible_child);
-		if (current != null) {
-			unowned var prev = current.prev;
-			if (current.prev != null) {
-				deck.visible_child = prev.data;
-				(current.data as Views.Base).unused = true;
-				Timeout.add (deck.transition_duration, clean_unused_views);
+	[GtkCallback]
+	void on_child_transition () {
+		if (deck.transition_running)
+			return;
+
+		bool after_current = false;
+		foreach (var child in deck.get_children ()) {
+			if (child == deck.visible_child) {
```

Comment:
`deck.get_adjacent_child (Hdy.NavigationDirection.FORWARD);`?

Reply:
Indeed I forgot we have that, that's even shorter that way.